### DISCUSSION TITLE
Differences between Devtools Fields and listed Backback fields

### DISCUSSION THREAD

**Discussion by garethwatt** [Laravel-Backpack/community-forum#800] [Q&A (Help)]
I'm new to backpack and started straight away with Devtools so I might be missing something.. 

When I create an ENUM field through Devtools the result is a text field in the admin interface instead of a dropdown as you would expect for something with defined values?? There aren't even any hints for what values are acceptable. In my case I defined gender to be "male" or "female"  (Not very progressive please don't cancel me)

The backpack 6 documentation for free fields says this about ENUM:
Show a select with the values for an ENUM database column, or an PHP enum (introduced in PHP 8.1).
Database ENUM
https://backpackforlaravel.com/docs/6.x/crud-fields#enum-1

There also seem to be other fields listed under backpack Free Fields that are missing or named differently in Devtools but I will start with this question first..

![Screen Shot 2024-01-15 at 11 12 14](https://github.com/Laravel-Backpack/community-forum/assets/23251630/e0edeae3-f725-4716-9d81-de0056e4241b)

I am using Laravel 10, Backpack 6 and MySQL

**Comment by karandatwani92** [comment]
Hi @garethwatt 

Yes, I understand you expected a dropdown. I did a check, It's not a devtool issue. It's an enhancement for the base package CRUD.
`CRUD::setFromDb();` is responsible for auto-selecting the column & field types.

I'm asking my colleague @pxpm to fix it by setting the Backpack field to `ENUM` type when it's an `ENUM` on the table. Currently, it is becoming a `text` field.⚠️

For now, you can manually turn it to [ENUM](https://backpackforlaravel.com/docs/6.x/crud-fields#enum-1) field in YourCRUDController to see the dropdown.

And... Devtools columns list & Backpack columns are different. Devtools is supposed to show DB column types.✅

Sorry for the inconvenience. Thanks for reporting it.🙏
